quasiquotes3/shared/src/main/scala/scala/meta/internal/quasiquotes/ExprLifts.scala

```diff
@@ -0,0 +1,513 @@
+package scala.meta
+package internal
+package quasiquotes
+
+import scala.runtime.ScalaRunTime
+import scala.quoted._
+import scala.meta.{Tree => MetaTree}
+import scala.meta.internal.trees.Quasi
+import scala.meta.internal.parsers.Absolutize._
+import scala.meta.internal.parsers.Messages
+
+import scala.collection.mutable
+import scala.annotation.tailrec
+
+// uses holes in implementation, for which we do not have access here, so we implement those elsewhere
+trait InternalTrait(using val internalQuotes: Quotes) {
+  def liftTree(tree: MetaTree): internalQuotes.reflect.Tree
+  def liftOptionTree[T: Type](maybeTree: Option[MetaTree]): internalQuotes.reflect.Tree
+  def liftTrees[T: Type](trees: Seq[MetaTree]): internalQuotes.reflect.Tree
+  def liftTreess(treess: List[List[MetaTree]]): internalQuotes.reflect.Tree
+  def liftQuasi0(quasi: Quasi, optional: Boolean = false): internalQuotes.reflect.Tree
+
+  protected def unquotesName(q: scala.meta.internal.trees.Quasi): Boolean
+}
+
+// TODO should be generated by a custom codegen script
+trait ExprLifts(using override val internalQuotes: Quotes)(isPatternMode: Boolean) extends ReificationMacros with InternalTrait {
```

Comment:
Was this generated previously? Why do we need so much more in Scala 3?

Reply:
It was all generated by a macro triggered inside ReificationMacros.scala scalameta/common/shared/src/main/scala/scala/meta/internal/trees/Liftables.scala (`materializeAst[]')